### PR TITLE
refactor user verification code, upgrade to Node 8 for async/await support

### LINKED ISSUE #197: move user whitelist and honorary members list to database
The FCC user whitelist being in the code seems like it could be a security issue. It would probably be better to store this list in the database.

### PR DESCRIPTION
**Pull request by no-stack-dub-sack**
This is a modification of @kolmstead's recent PR (#219) who laid the groundwork for this. This builds on top of his work to remove the whitelist and honorary member list out of the project's code and into the DB. Thanks Kent!

There is a **major breaking change here**, which is updating the project to Node 8 and utilizing `async/await` (which is the specific breaking change to older versions of Node). 

- upgrades project to Node 8 / NPM 5 
- utilizes `async/await` to allow for user verification service to be written in a more synchronous fashion. Removes need to nest axios promises inside nested Mongoose callbacks several layers deep.  
- removes old white-list helper file
- adds `engines` field to `package.json` file on the server to specify the version of Node to use to Heroku since Node 8 is the first stable version that supports `async/await`
- updates `CONTRIBUTING.md` to reflect new prerequisite of Node 8

@gwenf @smiller171 @bonham000 tagging you guys as well, since you all have been active in contributing to the project and since this is a relatively big change. Mostly just to make you each aware of the move to Node 8 (@gwenf I think you have already upgraded so should still run nicely on your machine), and second - since I have no experience with `async/await` previous to this, I figured this could use a review. I will note, though, that I have tested both the whitelist and honorary member list locally and that the user verification process is still working smoothly with this code. 

closes #197

### CODE REVIEW

File: server/routes/user.js [hunk 1]
```diff
@@ -32,111 +28,20 @@ router.post('/api/user', (req, res) => {
...
+    var isWhitelistedUser = !(username === req.body.username);
+    // if user is honorary member, they will be let in w/o certs
+    var isHonoraryMember = await checkHonoraryMemberList(username);
+    // process FCC verification...
+    isCertified(username, isHonoraryMember, isWhitelistedUser)
+    .then(certs => {
+      // update user with certs and correct status in DB
+      handleProcessedUser(certs, mongoId, req, res, username);
+    }).catch(err => console.log(err));
+  })().catch(err => console.log(err));
```
Comment (bonham000) [inline server/routes/user.js]: I think making this an IIFE makes it a little harder to see what's going on. Is there a need for the function `processUserVerification` at all? Can you just `await` `checkWhiteList` and `checkHonoraryMemberList` and make the overall Express callback `async`?

Comment (no-stack-dub-sack) [inline server/routes/user.js]: @bonham000 possibly, I'd have to think about that some

**Comment by no-stack-dub-sack** [comment]
@bonham000 per your suggestion, I updated the express callback to be async itself. This requires an extra handler function to make sure we're handling all exceptions though (as illustrated in [this article](http://thecodebarbarian.com/80-20-guide-to-async-await-in-node.js.html), under Darth Vader meme).

**Comment by smiller171** [comment]
Thanks for the heads up @no-stack-dub-sack. Since this is a major change I'll wait for it to be merged before I start working on #217. I have no opinion on the change itself.

**Comment by no-stack-dub-sack** [comment]
@smiller171 cool, yeah I would say that's a good idea. I started working on another major PR last night and realized how problematic that might get until I'm able to merge this in and rebase.

**Pull request merged**